### DISCUSSION TITLE
Matching eigenvalues and eigenvectors

### DISCUSSION THREAD

**Discussion by AChioquetta** [QuSpin/QuSpin#496] [Q&A]
Does the order of eigenvalues match the order of columns of the diagonalized matrix? I mean, if I choose the first column of the matrix, its corresponding eigenvalue will be the first value gave on the list of eigenvalues?  (I used eigh()to solve )

**Comment by JiahaoYao** [comment]
Hi @AChioquetta, thanks for asking the question! The `eigh()` function is a wrap on top of `numpy`, and you can refer to the [numpy website](https://numpy.org/doc/stable/reference/generated/numpy.linalg.eigh.html).


> w(…, M) ndarray
The eigenvalues in ascending order, each repeated according to its multiplicity.

**Comment by JiahaoYao** [comment]
However, i remember `eig()` does not preserve the order.
